### DISCUSSION TITLE
Is this binding using GDExtension ?

### DISCUSSION THREAD

**Discussion by Kiokiok** [godot-jolt/godot-jolt#175] [Questions] [answered]
If I build this using the instructions, can I drop it in my current project as a plugin, and swap the physics engine ? Or is it a source build of the engine ?
Thanks !

**Comment by mihe** [comment, marked as answer]
Yes, it uses GDExtension. It does not require any special build of Godot itself.

You'll have to forgive the lack of instructions on how to add this extension to a project. It isn't really in a usable state yet, so I haven't prioritized it, but basically you need three things:

1. A `godot-jolt.gdextension` file in your project folder that tells Godot where to look for the compiled extension binaries
2. Tell Godot that you want to use `JoltPhysics3D` as your physics engine, by setting it in your `project.godot` file. You can also do this in the project settings, under "Physics / 3D / Physics Engine", assuming you have "Advanced Settings" enabled.
3. Run a version of Godot that's compatible with the extension, which currently happens to be 4.0-beta16

You can find the first two in the examples project in this repository. [Here](https://github.com/godot-jolt/godot-jolt/blob/4d285e5a88d5f1595b04583affdf5f2842699a3b/examples/godot-jolt.gdextension) is the `.gdextension` file, and [here](https://github.com/godot-jolt/godot-jolt/blob/4d285e5a88d5f1595b04583affdf5f2842699a3b/examples/project.godot#L30-L32) is the change needed to your `project.godot` file.

Note that you'll probably want to change the file paths in that `godot-jolt.gdextension` file to better match your project's folder structure. Once this extension starts approaching some kind of alpha/beta stage then I'll provide precompiled binaries that bundle an already appropriate `godot-jolt.gdextension` file.

**Reply by Kiokiok** [reply]
Thank you for the quick reply ! I'll be on the lookout for a stable version ;)

**Reply by mihe** [reply]
It took some time, but the first [release](https://github.com/godot-jolt/godot-jolt/releases) is out, if you're still interested. :)
